src/EditorFeatures/CSharpTest/Formatting/Indentation/SmartIndenterEnterOnTokenTests.cs

```diff
@@ -1347,6 +1347,112 @@ void Main(object o)
+                indentationLine: 7,
+                expectedIndentation);
+        }
+
+        [Trait(Traits.Feature, Traits.Features.SmartIndent)]
+        [WpfTheory]
+        [InlineData("x", "is < 7 and (>= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is", "< 7 and (>= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is <", "7 and (>= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7", "and (>= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and", "(>= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (", ">= 3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>=", "3 or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3", "or > 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or", "> 50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or >", "50) or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50", ") or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50)", "or not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50) or", "not <= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50) or not", "<= 0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50) or not <=", "0;", 8)]
+        [InlineData("x is < 7 and (>= 3 or > 50) or not <= 0", ";", 8)]
+        public async Task IndentPatternsInFieldDeclarationCSharp9(string line1, string line2, int expectedIndentation)
+        {
+            var code = @$"
+class C
+{{
+    static int x = 7;
+    bool y = {line1}
+{line2}
+}}";
+            await AssertIndentNotUsingSmartTokenFormatterButUsingIndenterAsync(
+                code,
+                indentationLine: 5,
+                expectedIndentation);
+        }
+
+        [Trait(Traits.Feature, Traits.Features.SmartIndent)]
+        [WpfTheory]
+        [InlineData("<", "7 and (>= 3 or > 50) or not <= 0", 12)]
+        [InlineData("< 7", "and (>= 3 or > 50) or not <= 0", 12)]
+        [InlineData("< 7 and", "(>= 3 or > 50) or not <= 0", 12)]
+        [InlineData("< 7 and (", ">= 3 or > 50) or not <= 0", 12)]
+        [InlineData("< 7 and (>=", "3 or > 50) or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3", "or > 50) or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or", "> 50) or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or >", "50) or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or > 50", ") or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or > 50)", "or not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or > 50) or", "not <= 0", 12)]
+        [InlineData("< 7 and (>= 3 or > 50) or not", "<= 0", 12)]
+        [InlineData("< 7 and (>= 3 or > 50) or not <=", "0", 12)]
+        public async Task IndentPatternsInSwitchCSharp9(string line1, string line2, int expectedIndentation)
+        {
+            var code = @$"
+class C
+{{
+    void M()
+    {{
+        var x = 7;
+        var y = x switch
+        {{
+            {line1}
+{line2} => true,
```

Comment:
I'm not sure whether straight alignment (12, the actual value) is expected or one more indented (16) would be better. The current behavior (12) matches what happens in switches for regular boolean expressions, so having these similar patterns behave the same way seems fine for now to me.

Reply:
one more woudl be preferred.  but if you're matching boolean exprs that's ok.